webhooks/data-import-cron.go

```diff
@@ -0,0 +1,217 @@
+	if spec.PVC == nil && spec.Storage == nil {
+		return fmt.Errorf("missing DataVolume PVC or Storage fields")
+	}
+
+	if spec.PVC != nil && spec.Storage != nil {
+		return fmt.Errorf("duplicate storage definition, both target storage and target pvc defined")
+	}
+
+	if err := validateStorageSize(spec); err != nil {
+		return err
+	}
+
+	if err := validateStorageClassName(spec); err != nil {
+		return err
+	}
+
+	if spec.PVC != nil {
+		accessModes := spec.PVC.AccessModes
+		if len(accessModes) != 1 {
+			return fmt.Errorf("required value: exactly 1 access mode is required")
+		}
+		if err := validateAccessMode(accessModes[0]); err != nil {
+			return err
+		}
+		if spec.PVC.DataSource != nil || spec.PVC.DataSourceRef != nil {
+			return fmt.Errorf("external population is incompatible with DataImportCrons")
+		}
+	} else if spec.Storage != nil {
+		for _, accessMode := range spec.Storage.AccessModes {
+			if err := validateAccessMode(accessMode); err != nil {
+				return err
+			}
+		}
+		if spec.Storage.DataSource != nil || spec.Storage.DataSourceRef != nil {
+			return fmt.Errorf("external population is incompatible with DataImportCrons")
+		}
+	}
+
+	if err := validateNumberOfSources(spec.Source); err != nil {
+		return err
+	}
+
+	if registry := spec.Source.Registry; registry != nil {
+		if err := validateDataVolumeSourceRegistry(registry); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateStorageSize(spec *cdiv1beta1.DataVolumeSpec) error {
+	var name string
+	var resourceRequests core.ResourceList
+
+	if spec.PVC != nil {
+		resourceRequests = spec.PVC.Resources.Requests
+		name = "PVC"
+	} else if spec.Storage != nil {
+		resourceRequests = spec.Storage.Resources.Requests
+		name = "Storage"
+	}
+
+	storageSize, ok := resourceRequests["storage"]
+	if !ok {
+		return fmt.Errorf("%s size is missing", name)
+	}
+	if storageSize.Value() <= 0 {
+		return fmt.Errorf("%s size can't be equal or less than zero", name)
+	}
+	return nil
+}
+
+func validateStorageClassName(spec *cdiv1beta1.DataVolumeSpec) error {
+	var sc *string
+
+	if spec.PVC != nil {
+		sc = spec.PVC.StorageClassName
+	} else if spec.Storage != nil {
+		sc = spec.Storage.StorageClassName
+	}
+
+	if sc == nil || *sc == "" {
+		return nil
+	}
+
+	if scErrors := validation.IsDNS1123Subdomain(*sc); len(scErrors) > 0 {
+		return fmt.Errorf("invalid storage class name: %q", scErrors)
+	}
+	return nil
+}
+
+func validateAccessMode(accessMode core.PersistentVolumeAccessMode) error {
+	switch accessMode {
+	case core.ReadWriteOnce:
+	case core.ReadOnlyMany:
+	case core.ReadWriteMany:
+	case core.ReadWriteOncePod:
+	default:
+		return fmt.Errorf(`unsupported value: "%s": supported values: "ReadOnlyMany", "ReadWriteMany", "ReadWriteOnce", "ReadWriteOncePod"`, string(accessMode))
+	}
+	return nil
+}
+
+func validateNumberOfSources(source *cdiv1beta1.DataVolumeSource) error {
+	numberOfSources := 0
+	s := reflect.ValueOf(source).Elem()
+	for i := 0; i < s.NumField(); i++ {
```

Comment:
nit: Use `for i := range s.NumField()`

Reply:
Done.
TIL that `range` can have `int` argument. I assumed only slices and maps can be there.

Reply:
But `golangci-lint` complains about it:
```
golangci/golangci-lint info installed /workspace/bin/golangci-lint
/workspace/bin/golangci-lint run --timeout 5m
webhooks/data-import-cron.go:170:17: cannot range over s.NumField() (value of type int) (typecheck)
	for i := range s.NumField() {
	               ^
make: *** [Makefile:327: lint] Error 1
```

Even if it obviously works: https://go.dev/play/p/KbogyWbaifX?v=goprev

Reply:
IIRC that was introduced with golang 1.22. :)

Reply:
Maybe the problem is that the linter is already a bit old. IIRC, we are using `golangci-lint` 1.55.2, which is based in golang 1.21.3. Could you please try to update the `golangci-lint` version up to 1.62.2 and try again?

Reply:
I will do it in a separate PR.